cassiopeia/tools/small_parsimony.py

```diff
@@ -239,3 +241,235 @@ def score_small_parsimony(
                 "infer_ancestral_states=True."
             )
     return parsimony
+
+
+def fitch_count(
+    cassiopeia_tree: CassiopeiaTree,
+    meta_item: str,
+    root: Optional[str] = None,
+    infer_ancestral_states: bool = True,
+    state_key: str = "S1",
+):
+    """Runs the FitchCount algorithm.
+
+    Performs the FitchCount algorithm for inferring the number of times that
+    two two states transition to one another across all equally-parsimonious
+    solutions returned by the Fitch-Hartigan algorithm. The original algorithm
+    was described in Quinn, Jones, et al, Science (2021). The output is an 
+    MxM count matrix, where the values indicate the number of times that
+    m1 transitioned to m2 along an edge in a Fitch-Hartigan solution.
+    To obtain probabilities P(m1 -> m2), divide each row by its row-sum.
+
+    This procedure will only work on categorical data and will otherwise raise
+    an error.
+
+    Args:
+        cassiopeia_tree: CassiopeiaTree object with a tree and cell meta data.
+        meta_item: A column in the CassiopeiaTree cell meta corresponding to a
+            categorical variable.
+        root: Node to treat as the root.
```

Comment:
Does it mean that the tree is re-rooted at this node, or that only the subtree at this node will be considered?

Reply:
It means that only the subtree below this node is considered. I'll change this in the docs to make it more clear!